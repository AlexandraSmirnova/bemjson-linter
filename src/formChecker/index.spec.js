import { formLinter } from '.';
import {
    invalidSpaceJson1, invalidSpaceJson2, invalidSpaceJson3, invalidSpaceJson4,
    invalidIndentJson1, invalidIndentJson2, invalidHeaderJson1, invalidHeaderJson2, invalidHeaderJson3,
    invalidHeaderJson4, invalidHeaderJson5, invalidHeaderJson6, invalidFooterJson1, invalidFooterJson2,
    invalidFooterJson3, invalidFooterJson4, invalidFooterJson5, invalidFooterJson6, invalidSizeJson1,
    invalidSizeJson2, invalidSizeJson3, severalErrorsJson1, severalErrorsJson2
} from './__mocks__/invalidJsons';
import {
    validJson1, validJson2, validJson3, validJson4, validJson5, validJson6, validJson7, validJson8,
    validJson9, validJson10
} from './__mocks__/validJsons';

describe('test form linter', () => {
    test.each([
        invalidSizeJson1,
        invalidSizeJson2,
        invalidSizeJson3
    ])('should return error FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL for %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    })

    test.each([
        invalidSpaceJson1,
        invalidSpaceJson2,
    ])('should return error FORM.CONTENT_VERTICAL_SPACE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidSpaceJson3,
        invalidSpaceJson4,
    ])('should return error FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidIndentJson1,
        invalidIndentJson2,
    ])('should return error FORM.CONTENT_ITEM_INDENT_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidHeaderJson1,
        invalidHeaderJson2,
    ])('should return error FORM.HEADER_TEXT_SIZE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidHeaderJson3,
        invalidHeaderJson4,
    ])('should return error FORM.HEADER_VERTICAL_SPACE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidHeaderJson5,
        invalidHeaderJson6,
    ])('should return error FORM.HEADER_HORIZONTAL_SPACE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidFooterJson1,
        invalidFooterJson2,
    ])('should return error FORM.FOOTER_VERTICAL_SPACE__IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidFooterJson3,
        invalidFooterJson4,
    ])('should return error FORM.FOOTER_HORIZONTAL_SPACE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        invalidFooterJson5,
        invalidFooterJson6,
    ])('should return error FORM.FOOTER_TEXT_SIZE_IS_INVALID %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });


    test.each([
        severalErrorsJson1,
        severalErrorsJson2,
    ])('should return several errors %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toMatchSnapshot();
    });

    test.each([
        validJson1,
        validJson2,
        validJson3,
        validJson4,
        validJson5,
        validJson6,
        validJson7,
        validJson8,
        validJson9,
        validJson10,
    ])('should return array without errors for %#', (json) => {
        expect(formLinter(JSON.parse(json), json)).toEqual([]);
    });
});
