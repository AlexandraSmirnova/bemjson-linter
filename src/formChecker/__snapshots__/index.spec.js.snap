// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form linter should return error FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ контентного элемента должен задаваться
        с помощью модификатора space-h на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 8,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ контентного элемента должен задаваться
        с помощью модификатора space-h на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 8,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.CONTENT_ITEM_INDENT_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_ITEM_INDENT_IS_INVALID",
    "error": "Модификатора indent-b у элементов формы content-item должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 14,
        "line": 12,
      },
      "start": Object {
        "column": 13,
        "line": 7,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.CONTENT_ITEM_INDENT_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_ITEM_INDENT_IS_INVALID",
    "error": "Модификатора indent-b у элементов формы content-item должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 14,
        "line": 12,
      },
      "start": Object {
        "column": 13,
        "line": 7,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.CONTENT_VERTICAL_SPACE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ контентного элемента со значением space-v
        должен быть на 2 шага больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 8,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.CONTENT_VERTICAL_SPACE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ контентного элемента со значением space-v
        должен быть на 2 шага больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 8,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_HORIZONTAL_SPACE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 14,
      },
      "start": Object {
        "column": 9,
        "line": 10,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_HORIZONTAL_SPACE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 14,
      },
      "start": Object {
        "column": 9,
        "line": 10,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_TEXT_SIZE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_TEXT_SIZE_IS_INVALID",
    "error": "Размер текстовых блоков в подвале должен соответствовать эталонному",
    "location": Object {
      "end": Object {
        "column": 18,
        "line": 19,
      },
      "start": Object {
        "column": 17,
        "line": 14,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_TEXT_SIZE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_TEXT_SIZE_IS_INVALID",
    "error": "Размер текстовых блоков в подвале должен соответствовать эталонному",
    "location": Object {
      "end": Object {
        "column": 14,
        "line": 18,
      },
      "start": Object {
        "column": 24,
        "line": 13,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_VERTICAL_SPACE__IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 14,
      },
      "start": Object {
        "column": 9,
        "line": 10,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.FOOTER_VERTICAL_SPACE__IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.FOOTER_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 14,
      },
      "start": Object {
        "column": 9,
        "line": 10,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_HORIZONTAL_SPACE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 8,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_HORIZONTAL_SPACE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 8,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_TEXT_SIZE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_TEXT_SIZE_IS_INVALID",
    "error": "Все текстовые блоки в заголовке формы (элемент header) должны быть со значением
        модификатора size на 2 шага больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 14,
        "line": 12,
      },
      "start": Object {
        "column": 24,
        "line": 7,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_TEXT_SIZE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_TEXT_SIZE_IS_INVALID",
    "error": "Все текстовые блоки в заголовке формы (элемент header) должны быть со значением
        модификатора size на 2 шага больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 18,
        "line": 19,
      },
      "start": Object {
        "column": 17,
        "line": 14,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_VERTICAL_SPACE_IS_INVALID 0 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_VERTICAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 8,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.HEADER_VERTICAL_SPACE_IS_INVALID 1 1`] = `
Array [
  Object {
    "code": "FORM.HEADER_VERTICAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 8,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL for 0 1`] = `
Array [
  Object {
    "code": "FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL",
    "error": "Подписи и поля должны быть одного размера",
    "location": Object {
      "end": Object {
        "column": 2,
        "line": 17,
      },
      "start": Object {
        "column": 1,
        "line": 1,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL for 1 1`] = `
Array [
  Object {
    "code": "FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL",
    "error": "Подписи и поля должны быть одного размера",
    "location": Object {
      "end": Object {
        "column": 2,
        "line": 7,
      },
      "start": Object {
        "column": 1,
        "line": 1,
      },
    },
  },
]
`;

exports[`test form linter should return error FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL for 2 1`] = `
Array [
  Object {
    "code": "FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL",
    "error": "Подписи и поля должны быть одного размера",
    "location": Object {
      "end": Object {
        "column": 2,
        "line": 10,
      },
      "start": Object {
        "column": 1,
        "line": 1,
      },
    },
  },
  Object {
    "code": "FORM.INPUT_AND_LABEL_SIZES_SHOULD_BE_EQUAL",
    "error": "Подписи и поля должны быть одного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 8,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
]
`;

exports[`test form linter should return several errors 0 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ контентного элемента со значением space-v
        должен быть на 2 шага больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 12,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
  Object {
    "code": "FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ контентного элемента должен задаваться
        с помощью модификатора space-h на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 6,
        "line": 12,
      },
      "start": Object {
        "column": 16,
        "line": 3,
      },
    },
  },
]
`;

exports[`test form linter should return several errors 1 1`] = `
Array [
  Object {
    "code": "FORM.CONTENT_HORIZONTAL_SPACE_IS_INVALID",
    "error": "Горизонтальный внутренний отступ контентного элемента должен задаваться
        с помощью модификатора space-h на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 9,
      },
      "start": Object {
        "column": 9,
        "line": 4,
      },
    },
  },
  Object {
    "code": "FORM.FOOTER_VERTICAL_SPACE_IS_INVALID",
    "error": "Вертикальный внутренний отступ должен быть на 1 шаг больше эталонного размера",
    "location": Object {
      "end": Object {
        "column": 10,
        "line": 14,
      },
      "start": Object {
        "column": 9,
        "line": 10,
      },
    },
  },
]
`;
